src/lib/storage/fixed_string_dictionary_segment.cpp

```diff
@@ -56,8 +56,8 @@ ChunkOffset FixedStringDictionarySegment<T>::size() const {
 template <typename T>
 std::shared_ptr<BaseSegment> FixedStringDictionarySegment<T>::copy_using_allocator(
```

Comment:
Do we have a test that tests the counters in the context of a segment? I can only find `SegmentAccessCounterTest`.

Reply:
Partially. In EncodedSegmentTest you will find a few tests. They are in that test suite because I require almost all helper functions of that test. I should refactor those tests and helper functions. In my fork I already extracted the helper functions because I use them for a lot of others tests.

Reply:
What do you think is easier? I can add it to `SegmentsUsingAllocatorsTest.MigrateSegment`, but if you say that your refactored tests are a better place to test this (also using `get_supporting_segment_encodings_specs`), we can do it there.

Reply:
Implementing the test in SegmentsUsingAllocatorsTest.MigrateSegment seems totally valid to me.